keps/sig-apps/4212-declarative-node-maintenance/README.md

```diff
@@ -0,0 +1,2144 @@
+-  `.status.nodeStatuses[0].drainTargets` should be updated during a `Drain` stage. The drain
+  targets should be resolved according to the [Pod Selection](#pod-selection) and [Pod Selection and DrainTargets Example](#pod-selection-and-draintargets-example).
+-  `.status.nodeStatuses[0].drainMessage` should be updated during a `Drain` stage. The message
+   should be resolved according to [Pod Selection and DrainTargets Example](#pod-selection-and-draintargets-example).
+- `.status.nodeStatuses[0].PodsPendingEvacuation`, to indicate how many pods are left to start
+  evacuation from the first node.
+- `.status.nodeStatuses[0].PodsEvacuating`, to indicate how many pods are being evacuated from the first node.
+  These are the pods that have matching Evacuation objects. 
+- To keep track of the entire maintenance the controller will reconcile a `Drained` condition and
+  set it to true if all pods pending evacuation/termination have terminated on all target nodes
+  when drain is requested by the maintenance object.
+- NodeMaintenance condition or annotation can be set on the node object to advertise the current
+  phase of the maintenance.
+
+#### Supported Stage Transitions
+
+The following transitions should be validated by the API server.
+
+- Idle -> _Deletion_
+  - Planning a maintenance in the future and canceling/deleting it without any consequence.
+- (Idle) -> Cordon -> (Complete) -> _Deletion_.
+  - Make a set of nodes unschedulable and then schedulable again.
+  - The complete stage will always be run even without specifying it.
+- (Idle) -> (Cordon) -> Drain -> (Complete) -> _Deletion_.
+  - Make a set of nodes unschedulable, drain them, and then make them schedulable again.
+  - Cordon and Complete stages will always be run, even without specifying them.
+- (Idle) -> Complete -> _Deletion_.
+  -  Make a set of nodes schedulable.
+
+The stage transitions are invoked either manually by the cluster admin or by a higher-level
+controller. For a simple drain, cluster admin can simply create the NodeMaintenance with
+`stage: Drain` directly.
+
+### DaemonSet Controller
+
+The DaemonSet workloads should be tied to the node lifecycle because they typically run critical
+workloads where availability is paramount. Therefore, the daemonset controller should respond to
+the Evacuation only if there is a NodeMaintenance happening on that node and the DaemonSet is in
+the `drainTargets`. For example, if we observe the following NodeMaintenance:
+
+```yaml
+apiVersion: v1alpha1
+kind: NodeMaintenance
+...
+status:
+  nodeStatuses:
+    - nodeRef:
+        name: six
+      drainTargets:
+        - podPriority: 2147483647
+          podType: Default
+        - podPriority: 5000
+          podType: DaemonSet
+  ...
+```
+
+To register the daemonset controller as a controller evacuator, it should set the following
+annotation on its own pods.
+
+```yaml
+evacuation.coordination.k8s.io/daemonset.apps.k8s.io: "10000/controller"
```

Comment:
This KEP should make it clear that the implementer needs to register this annotation (or set of annotations).

Reply:
updated to:

```
To fulfil the Evacuation API, the daemonset controller should register itself as a controller
evacuator. To do this, it should ensure that the following annotation is present on its own pods.
```

is this  better?